### PR TITLE
[Core] Clean up math utils

### PR DESCRIPTION
**Pull request by loumalouomega**
This PR cleans up the MathUtils and add some template arguments for the sake of generality. Adding deprecated macro for duplicated methods

### CODE REVIEW

**Comment by pablobecker** [comment]
Looks good to me. Waiting for IC to say something (although i think it is not needed since these are not important changes)

File: kratos/utilities/math_utils.h [hunk 1]
```diff
@@ -464,243 +408,239 @@ class MathUtils
...
-     * @param InputMatrix Is the input matrix (unchanged at output)
-     * @param InvertedMatrix Is the inverse of the input matrix
-     * @param InputMatrixDet Is the determinant of the input matrix
+     * @brief It inverts matrices of order 2, 3 and 4
+     * @param rInputMatrix Is the input matrix (unchanged at output)
+     * @param rInvertedMatrix Is the inverse of the input matrix
+     * @param rInputMatrixDet Is the determinant of the input matrix
      */
-
+    template<class TMatrix1, class TMatrix2>
```
Comment (adityaghantasala) [inline kratos/utilities/math_utils.h:416]: Why are the these templates here ? to allow both sparse and dense matrices ??

File: kratos/utilities/math_utils.h [hunk 2]
```diff
@@ -464,243 +408,239 @@ class MathUtils
...
-     * @param InputMatrix Is the input matrix (unchanged at output)
-     * @param InvertedMatrix Is the inverse of the input matrix
-     * @param InputMatrixDet Is the determinant of the input matrix
+     * @brief It inverts matrices of order 2
+     * @param rInputMatrix Is the input matrix (unchanged at output)
+     * @param rInvertedMatrix Is the inverse of the input matrix
+     * @param rInputMatrixDet Is the determinant of the input matrix
      */
-
+    template<class TMatrix1, class TMatrix2>
```
Comment (adityaghantasala) [inline kratos/utilities/math_utils.h:487]: also here ?

File: kratos/utilities/math_utils.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/utilities/math_utils.h:416]: To allow bounded and unbounded

File: kratos/utilities/math_utils.h (hunk 2 above)
Comment (loumalouomega) [reply kratos/utilities/math_utils.h:487]: Idem, to allow bounded and unbounded

Comment (loumalouomega) [reply kratos/utilities/math_utils.h:487]: Of course can be used with sparse, but a sparse system with 4x4 is quite small

File: kratos/utilities/math_utils.h (hunk 1 above)
Comment (maceligueta) [reply kratos/utilities/math_utils.h:416]: I needed this! Thanks.

Comment (loumalouomega) [reply kratos/utilities/math_utils.h:416]: imagen

**Review by maceligueta — APPROVED** [review]

**Pull request merged**